src/fluree/db/peer/http_api.clj

```diff
@@ -275,7 +287,8 @@
           result (cond
                    (and (:cmd param) (:sig param))
                    (let [persist-resp (<? (fdb/submit-command-async conn param))
-                         result       (if (and (string? persist-resp) (-> param :txid-only false?))
+                         result       (if (and (string? persist-resp)
+                                               (not (default-to-false param :txid-only)))
```

Comment:
Maybe instead of a new fn?:
```
(if (true? (:txid-only param))
   persist-resp
   (<? (fdb/monitor-tx-async conn ledger persist-resp timeout)))
```

Reply:
Yeah that's way better. Do you think it's still important to check for the string value in `persist-resp`?

Reply:
I think so. Missed that in my response. If it isn't a string, I think it means there is an error, and with <? it should check/throw but maybe some other condition case exists that makes sense to leave the check in there for now bing this is a quick fix.